app/assets/javascripts/ransack_ui_jquery/index.js

```diff
@@ -1,2 +1,6 @@
 //= require ransack/predicates
-//= require ransack_ui_jquery/search_form
\ No newline at end of file
+//= require ransack_ui_jquery/search_form
+
+if(jQuery.fn.select2) {
+  window.Select2 = jQuery.fn.select2;
+}
```

Comment:
Im not sure why we are adding compatibility with Select2 version 4 to this branch since a lot of rework will be necessary to begin with.
 
Yes adding this hack will make it run with select2 version 3 and 4, but it won't be usable as is. Making this script work with v3 and v4 will be extensive work. 

I believe having a branch for each version would be the way to go forward and switch depending on select2 version running on app.